server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -556,6 +557,10 @@ boolean isDefaultContext() {
      * by the system itself rather than by a user action.
      */
     public void markAsSystemContext() {
```

Comment:
Please add javadoc that this method require additional security permissions from now on, thank you.

Reply:
Added to the javadoc here to describe that this is guarded by a permission and how to grant permission to use this.